Example/Sources/ConversationViewController.swift

```diff
@@ -315,9 +315,11 @@ extension ConversationViewController: MessagesDisplayDelegate {
 //        return .custom(configurationClosure)
     }
     
-    func configureAvatarView(_ avatarView: AvatarView, for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) {
+    func configureAvatarView(_ avatarView: UIImageView, for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) {
         let avatar = SampleData.shared.getAvatarFor(sender: message.sender)
```

Comment:
I just wanted to point out what I was trying to say before, this is an improvement to the example app but not MessageKit itself. This benchmark was done using your specific approach which is already possible in MessageKit.

We can't control what code users put inside of `configureAvatarView`. They're just going to use `avatarView.layer.cornerRadius`. 

I really don't like the `Factory` design pattern approach. I think it adds a lot of unnecessary boilerplate that the user may not use. If they want this pattern, they can create the factory themselves